src/rpc/misc.cpp

```diff
@@ -69,7 +70,11 @@ static RPCHelpMan validateaddress()
 
         UniValue detail = DescribeAddress(dest);
         ret.pushKVs(detail);
+    } else {
+        CHECK_NONFATAL(!error_msg.empty());
```

Comment:
nit, might be better like this, just realized I'm doing the same [here](https://github.com/bitcoin/bitcoin/pull/20391/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dR2415)
```diff
@@ -57,7 +57,8 @@ static RPCHelpMan validateaddress()
 {
     std::string error_msg;
     CTxDestination dest = DecodeDestination(request.params[0].get_str(), error_msg);
-    bool isValid = IsValidDestination(dest);
+    const bool isValid = IsValidDestination(dest);
+    CHECK_NONFATAL(isValid == error_msg.empty());
 
     UniValue ret(UniValue::VOBJ);
     ret.pushKV("isvalid", isValid);
@@ -71,7 +72,6 @@ static RPCHelpMan validateaddress()
         UniValue detail = DescribeAddress(dest);
         ret.pushKVs(detail);
     } else {
-        CHECK_NONFATAL(!error_msg.empty());
         ret.pushKV("error", error_msg);
     }
```